etc/sync-function-template.js

```diff
@@ -76,7 +76,20 @@ function synctos(doc, oldDoc) {
   var theDocType = getDocumentType(doc, oldDoc);
 
   if (isValueNullOrUndefined(theDocType)) {
-    throw({ forbidden: 'Unknown document type' });
+    if (isDocumentMissingOrDeleted(oldDoc) && isDocumentMissingOrDeleted(doc)) {
+      // Attempting to delete a document that does not exist. This may occur when bucket access/sharing
+      // (https://developer.couchbase.com/documentation/mobile/current/guides/sync-gateway/shared-bucket-access.html)
+      // is enabled and the document was deleted via the Couchbase SDK. Skip everything else and simply assign the
+      // public channel
+      // (https://developer.couchbase.com/documentation/mobile/current/guides/sync-gateway/channels/index.html#special-channels)
+      // to the document so that it can be replaced in the future.
```

Comment:
just so that I understand fully, _must_ the public channel be assigned here for the document to be replaced in the future?

Reply:
Actually, no, I misspoke (miswrote?). Assigning the public channel to the document makes it so that users will get a 404 Not Found if they attempt to fetch (i.e. "view") the deleted document rather than a 403 Forbidden. I'll update the comment in a small follow-up PR.